### PR TITLE
Source Airtable: make stream name unique; add id to names

### PR DESCRIPTION
**Pull request by artem1205**
## What
Resolving https://github.com/airbytehq/alpha-beta-issues/issues/774

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨

Stream names changed; schema update needed

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]

**Comment by artem1205** [comment]
Hi, @jnr0790 
As this PR contains breaking changes, I would ask you to communicate with customers so that they can update their source schema and reset their connections.

cc @lazebnyi @YowanR

**Comment by erica-airbyte** [comment]
@artem1205 is there a way to have this PR only effect NEW Airtable configurations and not affect current configurations.

**Comment by artem1205** [comment]
@erica-airbyte , I'll check it.

**Comment by artem1205** [comment]
@erica-airbyte , unfortunately it is not possible.
At the very beginning of `read` method, we call `discover` to dynamically update streams lists and get updated schemas.

**Comment by erica-airbyte** [comment]
@artem1205 please provide the following:

What is being updated/changed?
What will the customer see that is different?
Will this affect their destination data? 
What actions the customers need to take once the changes goes live? 

Thanks! cc @amaliaroye or @jnr0790 this might need to be coordinated next week.

**Comment by artem1205** [comment]
1. Stream names will be changed, 
2. e.g. `users/checkboxes` -> `users/checkboxes/tbl81WIAUZg5nwGN8`
3. Yes
4. refresh source schema will be needed (with resetting)

**Comment by amaliaroye** [comment]
Thank you for the clarification @artem1205! I think we should give customers at least 2 weeks to give them time to adjust their downstream queries.

**Comment by amaliaroye** [comment]
Hi @artem1205! Can we schedule this release for Monday, 3/20? I'll send out the outreach today, but want to give users time if they need to prepare new queries or anything in advance

**Comment by artem1205** [comment]
@amaliaroye , sure!

**Comment by artem1205** [comment]
@amaliaroye , Sure, I'll make a notification for myself.

**Comment by amaliaroye** [comment]
@artem1205 perfect!! Thank you so much!

File: airbyte-integrations/connectors/source-airtable/source_airtable/source.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def discover(self, logger: AirbyteLogger, config) -> AirbyteCatalog:
                     {
                         "stream_path": f"{base_id}/{table.get('id')}",
                         "stream": SchemaHelpers.get_airbyte_stream(
-                            f"{base_name}/{SchemaHelpers.clean_name(table.get('name'))}",
+                            f"{base_name}/{SchemaHelpers.clean_name(table.get('name'))}/{table.get('id')}",
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-airtable/source_airtable/source.py:56]: Why use the name at all, if the ID is the unique thing?

**Review by evantahler — COMMENTED** [review]
What's the upgrade path for the user?  Will their old catalogs (e.g. stream selections) be migrated or will they need to re-configure the source?  This might need to be a major version bump if this is a breaking change for syncs.

File: airbyte-integrations/connectors/source-airtable/source_airtable/source.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-airtable/source_airtable/source.py:56]: @evantahler , Customers will have to refresh source schema (with resetting).
ID is unique, but it is better to show "normal" name as well for user, e.g. [here](https://github.com/airbytehq/airbyte/pull/22704/files#diff-dfd0d62d60cf384182b1fd813d87208d94cca591ba4a12f8a5c68697a1356523R75) just using ID is not enough to distinguish tables.

Comment (artem1205) [reply airbyte-integrations/connectors/source-airtable/source_airtable/source.py:56]: version bumped.

**Review by evantahler — APPROVED** [review]
👍 code-wise.  It looks like you are coordinating with TCS (@amaliaroye)! 
Please be sure to use the breaking-change playbook - https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit#heading=h.r2xzw0kclodc

**Pull request merged**